integration_tests/test_swapping.py

```diff
@@ -0,0 +1,176 @@
+    render.run(renderapi.stack.delete_stack, source_stack_name)
+
+@pytest.fixture(scope='module')
+def target_stack(render, target_tspecs):
+    target_stack_name = "target_stack"
+    renderapi.stack.create_stack(target_stack_name, render=render)
+    renderapi.client.import_tilespecs(target_stack_name, target_tspecs, render=render)
+    renderapi.stack.set_stack_state(target_stack_name, 'COMPLETE', render=render)
+
+    yield target_stack_name
+    render.run(renderapi.stack.delete_stack, target_stack_name)
+
+@pytest.fixture(scope='module')
+def pt_matches1(render):
+    match_collection1 = "collection1"
+    pt_matches = [d for d in swap_pt_matches1]
+    renderapi.pointmatch.import_matches(match_collection1,
+                                        pt_matches,
+                                        render=render)
+    yield match_collection1
+    render.run(renderapi.pointmatch.delete_collection, match_collection1)
+
+
+@pytest.fixture(scope='module')
+def pt_matches2(render):
+    match_collection2 = "collection2"
+    pt_matches = [d for d in swap_pt_matches2]
+    renderapi.pointmatch.import_matches(match_collection2,
+                                        pt_matches,
+                                        render=render)
+    yield match_collection2
+    render.run(renderapi.pointmatch.delete_collection, match_collection2)
+
+
+def test_swap_z_module(render, source_stack, target_stack, source_tspecs, target_tspecs):
+    ex = {
+        "render": render_params,
+        "source_stack": [source_stack],
+        "target_stack": [target_stack],
+        "zValues": [[1015]],
+        "delete_source_stack": "False"
+    }
+
+    mod = swap_zs.SwapZsModule(input_data=ex, args=['--output_json', 'output.json'])
+    mod.run()
+
+    # assert that both the stack exist
+    list_of_stacks = renderapi.render.get_stacks_by_owner_project(host=render.DEFAULT_HOST,
+                                                                  port=render.DEFAULT_PORT,
+                                                                  owner=render.DEFAULT_OWNER,
+                                                                  project=render.DEFAULT_PROJECT,
+                                                                  render=render)
+    
+    assert(source_stack in list_of_stacks and target_stack in list_of_stacks)
+
+    source_tileids = [ts.tileId for ts in source_tspecs]
+    target_tileids = [ts.tileId for ts in target_tspecs]
+
+    tspecs1 = renderapi.stack.get_stack_tileIds(source_stack, render=render)
+    tspecs2 = renderapi.stack.get_stack_tileIds(target_stack, render=render)
+
+    assert(ts.tileId in target_tileids for ts in tspecs1)
+    assert(ts.tileId in source_tileids for ts in tspecs2)
+
+    ex['source_stack'] = [source_stack, target_stack]
+    ex['target_stack'] = [target_stack]
+
+    mod = swap_zs.SwapZsModule(input_data=ex, args=['--output_json', 'output.json'])
+    
+    with pytest.raises(RenderModuleException) as e:
```

Comment:
wasn't immediately obvious why this should fail.. i think because the target stack is both source and target.  can you add a comment?